Improved build project (#1850)
dotnet/SqlClient: BUILDGUIDE.md

>  msbuild -p:configuration=Release
> ```

> Only with package and NetStandardPackage reference types, a none-AnyCPU platform can be used, otherwise, the specified platform will be replaced with AnyCPU on the build process.

Reviewer: ```suggestion
A non-AnyCPU platform reference can only be used with package and NetStandardPackage reference types. Otherwise, the specified platform will be replaced with AnyCPU in the build process.
```
